components/centraldashboard/app/api.ts

```diff
@@ -23,13 +29,42 @@ interface CreateProfileRequest {
   user?: string;
 }
 
+// Shape of the AddContributorRequest body
+interface AddOrRemoveContributorRequest {
+  contributor?: string;
+}
+
 interface HasWorkgroupResponse {
+  user: string;
   hasAuth: boolean;
   hasWorkgroup: boolean;
 }
 
-const OPERATION_NOT_SUPPORTED = {
-  error: 'Operation not supported'
+export const roleMap = {
```

Comment:
I think you might be able to use an Enum here since it provides the reverse mapping automatically.

https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings

Reply:
So I completed the entire refactor, just to realize reverse translations are not supported for `string` valued `enums` leading to a larger implementation to get reverse translations by leveraging separately defined functions.

Reply:
I tried a few approaches since this comment but I keep running into type issues. I've reverted again, feel free to share a snippet that works for you

Reply:
You could try something like this:
```
type Role = 'admin'|'owner'|'edit'|'contributor';
const ROLES: ReadonlyMap<Role, Role> = new Map([
  ['admin', 'owner'],
  ['owner', 'admin'],
  ['edit', 'contributor'],
  ['contributor', 'edit'],
]);
```
Then, you can get just use something like `ROLES.get(n.roleRef.name as Role)`. To me, using the built-in types is always easier and lessens the test footprint.